File: 0507-perfect-number/main.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num % 2 == 1:
            return False

        divisors = []
        n = num // 2
        for divisor in range(1, n + 1):
            if num % divisor == 0:
                divisors.append(divisor)

        return sum(divisors) == num

File: 0507-perfect-number/main.c
#include <math.h>

bool checkPerfectNumber(int num) {
    if (num % 2 == 1) return 0;

    int total = 0;
    int n = ceil(sqrt(num));
    for (int i = 1; i < n; i++) {
        if (num % i == 0) {
            total += i;
            total += num / i;
        }
    }

    return total / 2 == num;
}
